forem/forem
# Allow admins to turn "some" hardcoded pages into a "Page"

## What type of PR is this? (check all applicable)


## Description
We need to give admins the ability to override the /code-of-conduct, /terms and /privacy with a "Pages" component. However, we do not want to make these pages default "Pages" components in the database because this will mean that we are not providing that Forem with the opportunity to follow along with any updates that we make to the hardcoded page (should they choose not to overwrite any information). 

Hence, the solution that the team has come up with is to show the the three pages in an "Override defaults" section with a warning that they will not receive updates should they override these pages. Should they till want to go ahed with overriding the page, this section will pre-populate the information in the new Page to provide a good user experience. 

## Related Tickets & Documents
https://github.com/forem/InternalProjectPlanning/issues/145

## QA Instructions, Screenshots, Recordings

1. Go to /admin/pages. You should see the overrides section containing 3 items.
<img width="1680" alt="Screenshot 2020-10-16 at 13 17 05" src="https://user-images.githubusercontent.com/2786819/96252291-035db400-0fb2-11eb-83cc-bd24d6e57e8a.png">

2. You will be able to click on the item to view and the override button to turn it into a new Page.
3. Click on override
4. You will see a "New Page" view open with pre-populated information like below:
<img width="1680" alt="Screenshot 2020-10-16 at 13 17 24" src="https://user-images.githubusercontent.com/2786819/96252298-08bafe80-0fb2-11eb-9b6f-397f2e7b9842.png">

5. Once you make changes to the description, you will see your new page display under the all "Pages" section where you will be able to edit it. You should also click on the view link to ensure that the updates have been processed successfully. 

<img width="1680" alt="Screenshot 2020-10-16 at 13 30 21" src="https://user-images.githubusercontent.com/2786819/96253395-cf838e00-0fb3-11eb-9e8b-f9a52db2c07b.png">
 
<img width="1680" alt="Screenshot 2020-10-16 at 13 31 08" src="https://user-images.githubusercontent.com/2786819/96253479-f346d400-0fb3-11eb-854e-63549737d0e4.png">

6. If you override all three pages you will see an information banner below:
<img width="1680" alt="Screenshot 2020-10-16 at 13 46 55" src="https://user-images.githubusercontent.com/2786819/96254817-33a75180-0fb6-11eb-9e04-b449a6b29638.png">

## Added tests?


## Added to documentation?

_ Still need to add it to gitbooks_ 

## [optional] Are there any post deployment tasks we need to perform?
None

## [optional] What gif best describes this PR or how it makes you feel?
![alt_text](https://media.giphy.com/media/l2JejvkIM7havSTPa/giphy.gif)
